src/mcts/search.cc

```diff
@@ -250,10 +249,12 @@ void Search::SendUciInfo() REQUIRES(nodes_mutex_) {
     } else if (score_type == "W-L") {
       uci_info.score = wl * 10000;
     }
-    const int l = static_cast<int>(std::round(500.0 * (1.0 - wl - d)));
-    // Using 1000-w-l instead of 1000*d for D score so that W+D+L add up to
-    // 1000.0.
-    uci_info.wdl = ThinkingInfo::WDL{w, 1000 - w - l, l};
+
+    const auto w = std::max(0, static_cast<int>(std::round(500.0 * (1.0 + wl - floatD))));
+    const auto l = std::max(0, static_cast<int>(std::round(500.0 * (1.0 - wl - floatD))));
+    // Using 1000-w-l so that W+D+L add up to 1000.0.
+    const auto d = std::max(0, 1000 - w - l);
```

Comment:
Maybe this? (ensures they always add up to 1000)
```cpp
auto d = 1000 - w - l;
if (d < 0) {
  w += d / -2;
  l = 1000 - w;
  d = 0;
}
```

Reply:
Isn't the requirement it should add up to 1000 artificial? E.g. for 1.4 + 1.4 + 997.2 this is hard, but 1 + 1 + 997, although only 999, is still "best" rounding IMHO...

Reply:
WDL is for chess UI, and all UIs that I saw (including visualizations/graphs that I made) assume w+d+l=1000.
It's actually pretty hard to avoid, usually it's some kind of stacked graph, there e.g. L and D is taken from the WDL output, and W takes all the remaining space.